File: Klanvision_backend/src/main/java/com/klanvision/backend/model/AdminUser.java
package com.klanvision.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Entity
@Table(name = "admins")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdminUser {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String password; 
    
    private String role = "ADMIN";
    private String status = "Offline";
    private String lastActive = "Never";
    private String color = "#6366F1";

    @ElementCollection
    private List<String> permissions;

    @com.fasterxml.jackson.annotation.JsonProperty("isAuthorized")
    private boolean isAuthorized = true;

    @com.fasterxml.jackson.annotation.JsonProperty("is2FAEnabled")
    private boolean is2FAEnabled = false;

    @com.fasterxml.jackson.annotation.JsonProperty("is2FAConfigured")
    private boolean is2FAConfigured = false;
    private String secret2FA;
    private int failed2FAAttempts = 0;
}

File: Klanvision_V5_Backend-master/models/AdminUser.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';
import { AdminUserPermission } from './AdminUserPermission.js';

const AdminUser = sequelize.define('AdminUser', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  username: {
    type: DataTypes.STRING,
    allowNull: false
  },
  email: {
    type: DataTypes.STRING,
    allowNull: false
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  password: {
    type: DataTypes.STRING,
    allowNull: false
  },
  role: {
    type: DataTypes.STRING,
    defaultValue: 'ADMIN'
  },
  status: {
    type: DataTypes.STRING,
    defaultValue: 'Offline'
  },
  color: {
    type: DataTypes.STRING,
    defaultValue: '#6366F1'
  },
  lastActive: {
    type: DataTypes.STRING,
    field: 'last_active',
    defaultValue: 'Never'
  },
  isAuthorized: {
    type: DataTypes.BOOLEAN,
    field: 'is_authorized',
    defaultValue: true
  },
  is2FAEnabled: {
    type: DataTypes.BOOLEAN,
    field: 'is2faenabled',
    defaultValue: true
  },
  is2FAConfigured: {
    type: DataTypes.BOOLEAN,
    field: 'is2faconfigured',
    defaultValue: false
  },
  secret2FA: {
    type: DataTypes.STRING,
    field: 'secret2fa'
  },
  failed2FAAttempts: {
    type: DataTypes.INTEGER,
    field: 'failed2faattempts',
    defaultValue: 0
  }
}, {
  tableName: 'admins',
  timestamps: false
});

// Setup relationship to auxiliary permissions join table
AdminUser.hasMany(AdminUserPermission, { foreignKey: 'admin_user_id', as: 'permissionAssociations', onDelete: 'CASCADE' });
AdminUserPermission.belongsTo(AdminUser, { foreignKey: 'admin_user_id' });

export default AdminUser;
export { AdminUser };
